File: binance4j-staking/src/main/java/com/binance4j/staking/dto/ProductPosition.java
package com.binance4j.staking.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Product position.
 * 
 * @param positionId            Staking position ID.
 * @param productId             Staking project ID.
 * @param asset                 Locked asset.
 * @param amount                Locked Amount.
 * @param purchaseTime          Subscription time.
 * @param duration              Lock period(days ).
 * @param accrualDays           Accrue days.
 * @param rewardAsset           Earned asset.
 * @param apy                   APY.
 * @param rewardAmt             Earned amount.
 * @param extraRewardAsset      Rewards assets of extra staking type.
 * @param extraRewardAPY        APY of extra staking type.
 * @param estExtraRewardAmt     Rewards of extra staking type, distribute when order expires.
 * @param nextInterestPay       Next estimated interest payment.
 * @param nextInterestPayDate   Next interest payment date.
 * @param payInterestPeriod     Interest cycle.
 * @param redeemAmountEarly     Early redemption amount.
 * @param interestEndDate       Interest accrual end date.
 * @param deliverDate           Redemption arrival time.
 * @param redeemPeriod          Redemption interval.
 * @param redeemingAmt          Amount under redemption.
 * @param partialAmtDeliverDate Arrival time of partial redemption amount of order.
 * @param canRedeemEarly        When it is true, early redemption can be operated.
 * @param renewable             When it is true, auto staking can be operated.
 * @param type                  Order type is auto-staking or normal.
 * @param status                Position status.
 */
@ApiModel("Product position.")
public record ProductPosition(@ApiModelProperty("Staking position ID.") String positionId, @ApiModelProperty("Staking project ID.") String productId,
		@ApiModelProperty("Locked asset.") String asset, @ApiModelProperty("Locked Amount.") String amount,
		@ApiModelProperty("Subscription time.") String purchaseTime, @ApiModelProperty("Lock period(days ).") String duration,
		@ApiModelProperty("Accrue days.") String accrualDays, @ApiModelProperty("Earned asset.") String rewardAsset, @ApiModelProperty("APY.") String apy,
		@ApiModelProperty("Earned amount.") String rewardAmt, @ApiModelProperty("Rewards assets of extra staking type.") String extraRewardAsset,
		@ApiModelProperty("APY of extra staking type.") String extraRewardAPY,
		@ApiModelProperty("Rewards of extra staking type, distribute when order expires.") String estExtraRewardAmt,
		@ApiModelProperty("Next estimated interest payment.") String nextInterestPay,
		@ApiModelProperty("Next interest payment date.") String nextInterestPayDate, @ApiModelProperty("Interest cycle.") String payInterestPeriod,
		@ApiModelProperty("Early redemption amount.") String redeemAmountEarly, @ApiModelProperty("Interest accrual end date.") String interestEndDate,
		@ApiModelProperty("Redemption arrival time.") String deliverDate, @ApiModelProperty("Redemption interval.") String redeemPeriod,
		@ApiModelProperty("Amount under redemption.") String redeemingAmt,
		@ApiModelProperty("Arrival time of partial redemption amount of order.") String partialAmtDeliverDate,
		@ApiModelProperty("When it is true, early redemption can be operated.") boolean canRedeemEarly,
		@ApiModelProperty("When it is true, auto staking can be operated.") boolean renewable,
		@ApiModelProperty("Order type is auto-staking or normal.") String type, @ApiModelProperty("Position status.") String status) {
}

File: connectors/src/main/kotlin/com/binance4j/staking/dto/ProductPosition.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.staking.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * Product position.
 *
 * @property positionId            Staking position ID.
 * @property productId             Staking project ID.
 * @property asset                 Locked asset.
 * @property amount                Locked Amount.
 * @property purchaseTime          Subscription time.
 * @property duration              Lock period(days ).
 * @property accrualDays           Accrue days.
 * @property rewardAsset           Earned asset.
 * @property apy                   APY.
 * @property rewardAmt             Earned amount.
 * @property extraRewardAsset      Rewards assets of extra staking type.
 * @property extraRewardAPY        APY of extra staking type.
 * @property estExtraRewardAmt     Rewards of extra staking type, distribute when
 *                              order expires.
 * @property nextInterestPay       Next estimated interest payment.
 * @property nextInterestPayDate   Next interest payment date.
 * @property payInterestPeriod     Interest cycle.
 * @property redeemAmountEarly     Early redemption amount.
 * @property interestEndDate       Interest accrual end date.
 * @property deliverDate           Redemption arrivar time.
 * @property redeemPeriod          Redemption interval.
 * @property redeemingAmt          Amount under redemption.
 * @property partialAmtDeliverDate Arrivar time of partial redemption amount of
 *                              order.
 * @property canRedeemEarly        When it is true, early redemption can be
 *                              operated.
 * @property renewable             When it is true, auto staking can be operated.
 * @property type                  Order type is auto-staking or normal.
 * @property status                Position status.
 */
@ApiModel("Product position.")
data class ProductPosition(
	@ApiModelProperty("Staking position ID.")
	var positionId: String,
	@ApiModelProperty("Staking project ID.")
	var productId: String,
	@ApiModelProperty("Locked asset.")
	var asset: String,
	@ApiModelProperty("Locked Amount.")
	var amount: String,
	@ApiModelProperty("Subscription time.")
	var purchaseTime: String,
	@ApiModelProperty("Lock period(days ).")
	var duration: String,
	@ApiModelProperty("Accrue days.")
	var accrualDays: String,
	@ApiModelProperty("Earned asset.")
	var rewardAsset: String,
	@ApiModelProperty("APY.")
	var apy: String,
	@ApiModelProperty("Earned amount.")
	var rewardAmt: String,
	@ApiModelProperty("Rewards assets of extra staking type.")
	var extraRewardAsset: String,
	@ApiModelProperty("APY of extra staking type.")
	var extraRewardAPY: String,
	@ApiModelProperty("Rewards of extra staking type, distribute when order expires.")
	var estExtraRewardAmt: String,
	@ApiModelProperty("Next estimated interest payment.")
	var nextInterestPay: String,
	@ApiModelProperty("Next interest payment date.")
	var nextInterestPayDate: String,
	@ApiModelProperty("Interest cycle.")
	var payInterestPeriod: String,
	@ApiModelProperty("Early redemption amount.")
	var redeemAmountEarly: String,
	@ApiModelProperty("Interest accrual end date.")
	var interestEndDate: String,
	@ApiModelProperty("Redemption arrivar time.")
	var deliverDate: String,
	@ApiModelProperty("Redemption interval.")
	var redeemPeriod: String,
	@ApiModelProperty("Amount under redemption.")
	var redeemingAmt: String,
	@ApiModelProperty("Arrivar time of partial redemption amount of order.")
	var partialAmtDeliverDate: String,
	@ApiModelProperty("When it is true, early redemption can be operated.")
	var canRedeemEarly: Boolean,
	@ApiModelProperty("When it is true, auto staking can be operated.")
	var renewable: Boolean,
	@ApiModelProperty("Order type is auto-staking or normal.")
	var type: String,
	@ApiModelProperty("Position status.")
	var status: String
)